### ISSUE TITLE
possibly speed up elmo pretraining/training

### ISSUE THREAD

**Issue by pitrack** [nyu-mll/jiant#319]
tl; dr this is all conjecture because I haven't profiled the code.

in certain cases, elmo pretraining can be sped up (by ignoring eval task scalars/embeddings until `train_for_eval`). The "fix" would be to load the `classifier_task_map` for just existing + `train_tasks`, and reload/refresh it at the start of `train_for_eval`.

if pytorch's computation graph were optimized, these operations might already be ignored. I haven't done profiling.

anyway, this is super low priority. only documenting it here (and will briefly mention in code comments) to note that it is possible to improve elmo scalar handling.

The best version would be if we can ignore unused elmo scalars in _all_ cases (even in `train_for_eval` and `eval`), which is possible if allennlp modifies their elmo implementation or if we write wrappers for almost all of the elmo-related allennlp classes.
